### ISSUE TITLE
styled-jsx: plugins is not a function

### ISSUE THREAD

**Issue by M1ck0** [vercel/next.js#23293]
### What version of Next.js are you using?

10.0.9

### What version of Node.js are you using?

15.11.0

### What browser are you using?

Chrome, Brave, Firefox

### What operating system are you using?

Linux, WIndows, MacOS

### How are you deploying your application?

next start

### Describe the Bug

When trying to set up styled-jsx to work with sass I get the following error

```
event - build page: /
wait  - compiling...
info  - Using external babel configuration from /home/m1ck0/Desktop/projects/test-next-style/with-styled-jsx-scss-app/.babelrc
error - ./pages/index.js
TypeError: /home/m1ck0/Desktop/projects/test-next-style/with-styled-jsx-scss-app/pages/index.js: plugins is not a function
event - build page: /next/dist/pages/_error
wait  - compiling...
error - ./pages/index.js
TypeError: /home/m1ck0/Desktop/projects/test-next-style/with-styled-jsx-scss-app/pages/index.js: plugins is not a function
Error: Cannot find module '/home/m1ck0/Desktop/projects/test-next-style/with-styled-jsx-scss-app/.next/server/pages-manifest.json'
```

### Expected Behavior

Project should build with no problem

### To Reproduce

Add this to `.babelrc`
```js
{
  "presets": [
    [
      "next/babel",
      {
        "styled-jsx": {
          "plugins": ["styled-jsx-plugin-sass"]
        }
      }
    ]
  ]
}
```

Add this to `pages/index.js`

```jsx
const backgroundColor = '#eee'

export default function Home() {
  return (
    <div className="hello">
      <p>Hello World</p>
      <style jsx>{`
        $color: red;

        .hello {
          background-color: ${backgroundColor};
          padding: 100px;
          text-align: center;
          transition: 100ms ease-in background;

          &:hover {
            color: $color;
          }

          @media only screen and (max-width: 480px) {
            font-size: 20px;
          }
        }
      `}</style>
    </div>
  )
}
```

**Comment by fabien** [comment]
@M1ck0 I'm getting this bug as well. Why did you close this issue?

**Comment by M1ck0** [comment]
Use [@styled-jsx/plugin-sass](https://www.npmjs.com/package/@styled-jsx/plugin-sass) instead of `styled-jsx-plugin-sass` in `.babelrc` then tell me if it works. I'll reopen it for now until you resolve it.

**Comment by fabien** [comment]
@M1ck0 thanks, it works using `@styled-jsx/plugin-sass`! Shouldn't we keep this open so it will be fixed in the examples?

**Comment by M1ck0** [comment]
Later I'll send PR for this issue, but you are right. I'll leave this open until that PR is merged.

**Comment by hpachy** [comment]
Hello guys,

I running into the same problem... with you solution I compile without any problem so that's fine. but now an other problem wich is the style does not appear at all... And before I changed my .babelsrc file (old config or no one) It was more weird because I had the same problem but after a reload on the page the style loaded...

I'm very confused and I don't know how to fix it...

**Comment by M1ck0** [comment]
Did you try to build your project? If so is it still not showing up?

**Comment by hpachy** [comment]
Yes I did `yarn run build` and I deployed it on vercel and nothing to do the style tag do not exist.... 
my following code : 

```js
    var style = '.toto {background: red}'

    <div className="toto">
      <div>
        toto
      </div>
      <style global jsx>
        {`
         ${style}
        `}
      </style>
    </div>
```

**Comment by M1ck0** [comment]
Waiting for #23414 to be merged

**Comment by hpachy** [comment]
how could I fix the problem with out waiting the PR to be approved ?

**Comment by M1ck0** [comment]
> Yes I did `yarn run build` and I deployed it on vercel and nothing to do the style tag do not exist....
> my following code :
> 
> ```js
>     var style = '.toto {background: red}'
> 
>     <div className="toto">
>       <div>
>         toto
>       </div>
>       <style global jsx>
>         {`
>          ${style}
>         `}
>       </style>
>     </div>
> ```

You mean this problem?

**Comment by hpachy** [comment]
yes indeed

**Comment by M1ck0** [comment]
Could you make a github repo with the issue so I can inspect it futher?

**Comment by hpachy** [comment]
@M1ck0  I invited you to one of my private github repo to sharing you the code currently not working on vercel... but working fine in the localhost :/

**Comment by M1ck0** [comment]
I see.. Let me check. I'll get back to you soon.

**Comment by M1ck0** [comment]
All styles that are added with `styled-jsx` do exist in View Page Source, which means that they are applied as they should be. One thing that might be a problem is that you are using 3 ways to style an app. `.css`, `.module.css` and `styled-jsx` so they might be overriding each-other when in production.

It does work on localhost but while in `dev` mode only.

You should try to make a new project and see if `styled-jsx` works well.

Also, you don't need to install `styled-jsx` package as it is shipped with `NextJS` by default.

**Comment by hpachy** [comment]
I not really sure that because when I look at the head tag I don't see any 
>jsx style
<img width="352" alt="Capture d’écran, le 2021-03-30 à 15 21 10" src="https://user-images.githubusercontent.com/24634810/112995797-e0901800-916b-11eb-9991-2cf715e2123b.png">
instead of the local host which they are implicitly writen
<img width="1286" alt="Capture d’écran, le 2021-03-30 à 15 20 50" src="https://user-images.githubusercontent.com/24634810/112995812-e5ed6280-916b-11eb-9614-7c9caa7bb58c.png">

I trying to delete all the style, but I'm not sure that's the solution :/

**Comment by hpachy** [comment]
So I come back to figured out my problem so it's because styled-jsx  compiles styles at build time [712](https://github.com/vercel/styled-jsx/issues/712#issuecomment-810886891)

**Comment by robertolemesp** [comment]
> 
> 
> So I come back to figured out my problem so it's because styled-jsx compiles styles at build time [712](https://github.com/vercel/styled-jsx/issues/712#issuecomment-810886891)
> 
> > Plugins make it possible to use popular preprocessors like SASS, Less, Stylus, PostCSS or apply custom transformations to the styles at compile time.

i have solved this issue installing node-sass and then deploying to vercel

**Comment by robertolemesp** [comment]
> 
> 
> @M1ck0 I invited you to one of my private github repo to sharing you the code currently not working on vercel... but working fine in the localhost :/

i have solved an similar issue when deploying to vercel ("plugins is not a function") just by installing node-sass

**Comment by robertolemesp** [comment]
> 
> 
> @M1ck0 I invited you to one of my private github repo to sharing you the code currently not working on vercel... but working fine in the localhost :/

i have solved an similar issue when deploying to vercel ("plugins is not a function") just by installing node-sass

**Comment by M1ck0** [comment]
> > @M1ck0 I invited you to one of my private github repo to sharing you the code currently not working on vercel... but working fine in the localhost :/
> 
> i have solved an similar issue when deploying to vercel ("plugins is not a function") just by installing node-sass

The issue will be resolved with #23414 pull request

**Comment by icey0126** [comment]
i met a similar error message while i using  styled-jsx in my project. Then i found the reason is that my node version is too low. Hope can help someone.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
